processor/groupbytraceprocessor/event_test.go

```diff
@@ -106,6 +107,38 @@ func TestEventCallback(t *testing.T) {
 	}
 }
 
+func TestOnTraceReleasedAsync(t *testing.T) {
+	waitCh := make(chan struct{})
+	defer close(waitCh)
+
+	em := newEventMachine(zap.NewNop(), 10)
+	em.onTraceReleased = func([]pdata.ResourceSpans) error {
+		<-waitCh
+		return nil
+	}
+
+	em.startInBackground()
+	defer em.shutdown()
+
+	var batchReceived int32
+	em.onBatchReceived = func(traces pdata.Traces) error {
+		atomic.StoreInt32(&batchReceived, 1)
```

Comment:
Why not a WaitGroup instead?

Reply:
I don't really understand the idea with WaitGroup. If you offer making wg, incrementing in line 123, and doing `wg.Done()` in 125, waiting at the end of the test, then I didn't do it on purpose - in case of test failure it will block forever. In my opinion its better make a non-blocking test for sure than keeping github actions warm :)

Reply:
In this part, you'd use `wg.Add(1)`, and use `wg.Wait()` instead of waiting a magic number of milliseconds, which might not be enough in future runs.

Reply:
Then this test blocks forever in the fail case that is tests. I see problems in both approaches, the question is what is less painful?

Reply:
It will block "forever" only when there's a real problem with the test subject, as opposed to random failures caused by bad timing.

Reply:
Removed the test, because we decided not to change eventmachine